contracts/ToucanRegenBridge.sol

```diff
@@ -178,6 +178,8 @@ contract ToucanRegenBridge is Ownable, Pausable, AccessControl {
         require(!origins[origin], "duplicate origin");
         origins[origin] = true;
 
+        require((totalTransferred - amount) >= 0, "balance underflow");
```

Comment:
This check is not required, Solidity 0.8+ handles underflow issues. Btw can you add unit tests to cover these cases (underflow issue plus proper totalTransferred updates)?

Reply:
Yes, you are right.  I just added this check to avoid get the generic solidity message what could lead user have a misunderstanding reason why the function is failing.  What do you think, should I remove or leave the check?

Solidity default message for underflow:
`Note: The called function should be payable if you send value and the value you send should be less than your current balance.`

Reply:
This check will actually underflow before the the `>=` check resolves because `totalTransferred - amount` is a `uint256` hence Slither figured out that you have a tautology (`uint256s` are always >= 0). I don't mind the built-in error, we are already leveraging it in the `tco2Limits` check just below and there is a test case that covers that [here](https://github.com/regen-network/toucan-bridge/blob/5d681b040ff1be71ba5cee51aa3a5f6d5adee47c/test/ToucanRegenBridge.js#L239).

Reply:
ok, got it! 😀  Tks @0xmichalis